lib/opal/builder/scheduler.rb

```diff
@@ -15,7 +15,7 @@ def initialize(builder)
 
   singleton_class.attr_accessor :builder_scheduler
 
-  if RUBY_ENGINE != 'opal'
+  if RUBY_ENGINE == 'ruby'
```

Comment:
I expect to implement a threaded scheduler for jruby (and maybe truffleruby), thats why i have 'ruby' here, because i expect to add a `elsif RUBY_ENGINE == 'jruby'` and maybe one or two others more, can't say yet, need more research. Please merge as is, will address this soon with PR for threaded scheduler after PR for truffleruby.

Reply:
What I mean by this is that we will add prefork to be compiled into bundle whenever anyone requires `opal-parser`. Which is why, even if code looks a little bit worse, it's good to keep it as is.

Reply:
I will address this at a later stage

Reply:
Addressed in #2688

Reply:
See https://github.com/opal/opal/blob/ba4c7851766919c048e43fd2aae744982c6ecc5a/lib/opal/builder/scheduler.rb#L66